toponetx/datasets/mesh.py

```diff
@@ -92,37 +91,37 @@ def shrec_16(size="full"):
     url = "https://github.com/mhajij/shrec_16/raw/main/shrec.zip"
     url_small = "https://github.com/mhajij/shrec_16/raw/main/small_shrec.zip"
     if size == "full":
```

Comment:
We have some duplicated code here, which adds unnecessary complexity. I'd suggest grouping the differences into one if/elif and then only having the download / extract logic once:
```python
    if size == "full":
        url = url
        zipfile = os.path.join(DIR, "shrec.zip")
        training = os.path.join(DIR, "shrec_training.npz")
        testing = os.path.join(DIR, "shrec_testing.npz")
    elif size == "small":
        url = url_small
        zipfile = os.path.join(DIR, "small_shrec.zip")
        training = os.path.join(DIR, "small_shrec_training.npz")
        testing = os.path.join(DIR, "small_shrec_testing.npz")
    else:
        raise ValueError(f"size must be 'full' or 'small' got {size}.")
    if not os.path.isfile(zipfile):
    ...
```

Could be further reduced to something like:
```python
    DS_MAP = {
        'full': ("shrec", "https://github.com/mhajij/shrec_16/raw/main/shrec.zip"),
        'small': ("small_shrec", "https://github.com/mhajij/shrec_16/raw/main/small_shrec.zip")
    }
    if size not in DS_MAP:
        raise ValueError(f"size must be 'full' or 'small' got {size}.")
    url, ds_name = DS_MAP[size]
    
    zipfile = os.path.join(DIR, f"{ds_name}.zip")
    training = os.path.join(DIR, f"{ds_name}_training.npz")
    testing = os.path.join(DIR, f"{ds_name}_testing.npz")
```

Reply:
I think the second option is better since it is shorter and shows the naming logic on the first look.

Reply:
thanks, yes it is much better